rfcs/0000-central-station.md

```diff
@@ -0,0 +1,507 @@
+process by concentrate.
+
+### With CentralStation
+
+Busloc will be pared down to something more Trike-esque. Since on-premise Windows servers are so
+hard to deal with, the smaller Busloc is the better. It will continue to handle the TransitMaster
+integration, but it will put this data into Kinesis for CentralStation to consume.
+
+The Samsara feeds will be removed, as those will go directly to CentralStation.
+
+## Trainloc / CommuterRailBoarding
+
+### Current description
+
+TrainLoc and CommuterRailBoarding are two applications that once were separate but were
+consolidated into a single umbrella application, using the name Commuter Rail Boarding, and they
+manage the realtime data of commuter rail trains from Keolis.
+
+Keolis offers two firebase feeds, consumed by CRB:
+
+- The "boarding status" feed, which we get for trains at a few stations, with info like "now
+  boarding on track 9". We put these in TripUpdates_enhanced.json, in our extension
+  `boarding_status` field. Concentrate generally merges these with predictions for the trains, but
+  we often end up with StopTimeUpdates with null arrivals and departures.
+- The vehicle positions feed. We get updates every ~15 seconds with the position of every commuter
+  rail train. If the AVL equipment is working and the train is logged in, it includes the GTFS
+  trip short name. Otherwise we get lat/long and a vehicle ID, but not the trip.
+
+The VehiclePositions are consumed by Swiftly to produce our predictions. The trains without a trip
+short name get auto-assigned a GTFS trip by Swiftly, and Keolis customer service people log into
+the Swiftly dashboard to manually re-tag trains that have been given an incorrect assignment.
+
+### With CentralStation
+
+The vehicle positions feed will go straight to CentralStation. At least initially, the boarding
+status feed will continue to be handled by CRB and continue to go to concentrate as TripUpdates.
+In other words CRB will become CRB again, and drop TrainLoc entirely.
+
+## Glides
+
+### Current description
+
+Right now Glides is just a consumer of the realtime data, directly from RTR.
+
+### With CentralStation
+
+Glides will either get all its data from CentralStation, or get the public data from our public
+feeds, and supplement it with private data from CentralStation.
+
+It will use CentralStation's API for train tagging and service changes.
+
+## Skate
+
+### Current description
+
+Skate consumes bus realtime data, straight from Busloc. This data is GTFS-rt-esque, enriched with
+operator information and block and run data.
+
+### With CentralStation
+
+Skate will get all its vehicle position data from public feeds. It will supplement it with private
+data from CentralStation.
+
+Depending on if we want CentralStation to store PI operator data, it may continue to get that
+information from an S3 file produced by Busloc.
+
+It will use CentralStation's API for train tagging and service changes.
```

Comment:
I assume you mean bus tagging? And in this future world, are we imagining that associating a bus to a trip would still be roughly as automatic as with Swiftly, just with a manual override?